### PR TITLE
Roughly one fifth of internal affairs agents are traitors with an extra objective

### PR DESCRIPTION
**Pull request by 4dplanner**
:cl:
add: 20% of internal affairs agents are actually traitors
/:cl:

Robustin suggested this, and it does add a bit more paranoia

### CODE REVIEW

**Comment by Robustin** [comment]
I'm curious, I saw a round yesterday where it was roughly 30% traitors and 70% IAA. I wasn't really paying attention that round but it appeared to work well since there was enough traitor activity to mask the IAA activity and prevent metagaming. I assumed it was added as a feature but if it wasn't, then at least an admin had the right idea (no admins were on at the end of the round).

This always felt like a no-brainer decision. We wanted DA to be kind of like traitor, but more focused on antag intrigue. Mixing IAA's which are barely even antags, with actual traitors, solves a lot of the mode's problems and makes it into something that could be a regular in rotation (and justify simpler rules for IAA's since we can have enough traitors to create chaos).

**Comment by 4dplanner** [comment]
Are you sure? The only IAA round I saw running had 100% agents, so I'm assuming that was admin intervention

**Comment by 4dplanner** [comment]
And considering IAA can only be admin run at the moment it doesn't seem too unlikely

File: code/game/gamemodes/traitor/double_agents.dm [hunk 1]
```diff
@@ -238,6 +246,11 @@
 		state.add_steal_targets_timer()
 		if(!issilicon(traitor.current))
 			give_pinpointer(traitor)
+		//Optional traitor objective
+		if(prob(PROB_ACTUAL_TRAITOR))
+			traitor.special_role = TRAITOR_AGENT_ROLE
+			traitor.licence_to_kill = TRUE
```
Comment (RemieRichards) [inline code/game/gamemodes/traitor/double_agents.dm]: this var isn't needed, just check for ```traitor.special_role == TRAITOR_AGENT_ROLE```

Comment (4dplanner) [inline code/game/gamemodes/traitor/double_agents.dm]: That's how I did it originally, but I thought this way would be better to allow for an easier transition to a non role-based framework (IAAchan when). Nonetheless, I'll revert it

**Pull request merged**